Tc 84

lukezhang454 opened the pull request:
Initial stripe integration

I added an express server to handle stripe backend calls. You can run both the application and the server using:

`npm run dev`

CyberKronos commented:
1. Move checkoutForm styling from `App.css` to `/semantic/src/site/globals/site.overrides`. We should try to move all styling into the semantic theming structure. Will delete App.css soon.

2. CheckoutForm.jsx
- instead of using the local react state, can we leverage redux instead? Store needed state into redux store. This way we can just dispatch fewer actions, instead of setState a bunch of times.
- `convertToStripePrice` can be moved out to the `utils` folder
- move inline styling into the correct `.less` files
- Use semantic components where necessary (i.e. <h1 style={{ ... }}> can be replaced with semantic <Header /> component)

3. /ducks/.../packs
- Lint everything in `operations.js`

lukezhang454 commented:
Addressed points 1, 3 and created new checkout reducer for handling state changes.

TODOs:
* read `customerId` from `users` state instead of defining it separately under CheckoutForm
* `convertToStripePrice` can be moved out to the `utils` folder
* move inline styling into the correct `.less` files
* Use semantic components where necessary (i.e. <h1 style={{ ... }}> can be replaced with semantic  component)

CyberKronos commented:
Merging existing work done to master. Open new PR(s) with the rest of the changes.
